test/fixtures/basic/types.ts

```diff
@@ -246,7 +246,7 @@ describe('app config', () => {
       nested: {
         val: number
       },
-      userConfig: number
+      userConfig?: number
```

Comment:
Is it correct that this is possibly undefined? We have both a default value in schema and also (when merged) also a value in user-provided config that we know about.

Reply:
Schema is partial, so yes this is _expected_ (modules might or might not finally provide the schema values. and we do not load defaults from schema)

 But if you know a way to make it required when user provides value, feel free to push it :)